src/main/java/uk/gov/hmcts/reform/pip/data/management/database/AzureBlobService.java

```diff
@@ -69,19 +74,31 @@ public String uploadFlatFile(String payloadId, MultipartFile file) {
      * @return the data contained within the blob in String format.
      */
     public String getBlobData(String payloadId) {
-        BlobClient blobClient = blobContainerClient.getBlobClient(payloadId);
+        BlobClient blobClient = artefactBlobContainerClient.getBlobClient(payloadId);
         return blobClient.downloadContent().toString();
     }
 
     public Resource getBlobFile(String payloadId) {
-        BlobClient blobClient = blobContainerClient.getBlobClient(payloadId);
+        BlobClient blobClient = artefactBlobContainerClient.getBlobClient(payloadId);
         byte[] data = blobClient.downloadContent().toBytes();
         return new ByteArrayResource(data);
     }
 
     public String deleteBlob(String payloadId) {
-        BlobClient blobClient = blobContainerClient.getBlobClient(payloadId);
+        BlobClient blobClient = artefactBlobContainerClient.getBlobClient(payloadId);
         blobClient.delete();
         return String.format(DELETE_MESSAGE, payloadId);
     }
+
+    /**
+     * Delete a blob in the publications storage container that contain the artefact ID.
+     *
+     * @param blobName The name of the blob to delete.
+     * @return A delete message.
+     */
+    public String deletePublicationBlob(String blobName) {
+        BlobClient blobClient = publicationsBlobContainerClient.getBlobClient(blobName);
+        blobClient.delete();
```

Comment:
So for an expired blob uploaded before we implement this publication blob storage, it won't have any PDF or Excel. What is the value of blobClient in this case? If it is null, it will cause a null pointer exception when we call the delete method

Reply:
JSON blobs are not being used in production currently so shouldn't occur. This deletion works exactly the same as how artefact/flat files are currently being deleted.